src/App.jsx

```diff
@@ -3,35 +3,48 @@ import React, { useState, useEffect } from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
+function App() {
+  //const Is = JSON.parse(localStorage.getItem('todoList')) ?? [];
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('todoList')) ?? []);
+  const [isLoading, setIsLoading] = useState(true);
 
+  // First effect useEffect to load the initial data
   useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-}
-
-function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+    const promise = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({ data: { todoList } });
```

Comment:
Nice job implementing this useEffect to mock a async call! One thing to be on the look out for. Per the instructions, you want `JSON.parse(localStorage.getItem('todoList')) ?? []` In place of `todoList` and for your `useState` parameter you actually want `[]` an empty array. This seems to be working just fine since your 'uesEffect` is called on render but if there was ever a problem with your `useEffect` this could cause some issues.